packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -469,42 +474,44 @@ export class TSRHandler {
 	private _triggerUpdateDevices() {
 		if (!this._initialized) return
 
-		if (this._triggerUpdateDevicesTimeout) clearTimeout(this._triggerUpdateDevicesTimeout)
+		if (this._triggerUpdateDevicesTimeout) {
+			clearTimeout(this._triggerUpdateDevicesTimeout)
+		}
 		this._triggerUpdateDevicesTimeout = undefined
 
-		if (!this._updateDevicesIsRunning) {
-			this._updateDevicesIsRunning = true
-			debug('triggerUpdateDevices now')
-
-			// Defer:
-			setTimeout(() => {
-				this._updateDevices().then(
-					() => {
-						this._updateDevicesIsRunning = false
-						if (this._triggerUpdateDevicesCheckAgain) {
-							debug('triggerUpdateDevices from updateDevices promise resolved')
-							if (this._triggerUpdateDevicesTimeout) clearTimeout(this._triggerUpdateDevicesTimeout)
-							this._triggerUpdateDevicesTimeout = setTimeout(() => this._triggerUpdateDevices(), 1000)
-						}
-						this._triggerUpdateDevicesCheckAgain = false
-					},
-					() => {
-						this._updateDevicesIsRunning = false
-						if (this._triggerUpdateDevicesCheckAgain) {
-							debug('triggerUpdateDevices from updateDevices promise rejected')
-							setTimeout(() => this._triggerUpdateDevices(), 1000)
-							if (this._triggerUpdateDevicesTimeout) clearTimeout(this._triggerUpdateDevicesTimeout)
-							this._triggerUpdateDevicesTimeout = setTimeout(() => this._triggerUpdateDevices(), 1000)
-						}
-						this._triggerUpdateDevicesCheckAgain = false
-					}
-				)
-			}, 10)
-		} else {
-			// oh, it's already running, cue a check again later:
+		if (this._updateDevicesIsRunning) {
 			debug('triggerUpdateDevices already running, cue a check again later')
 			this._triggerUpdateDevicesCheckAgain = true
+			return
 		}
+		this._updateDevicesIsRunning = true
+		debug('triggerUpdateDevices now')
+
+		// Defer:
+		setTimeout(() => {
+			this._updateDevices()
+				.then(
+					() =>
+						this._triggerUpdateDevicesCheckAgain &&
```

Comment:
What does this check do? It seems to be a leftover from the refactor. 
It seems that it makes it so we don't log anything if this is set to false, which is not what the name of the variable indicates.

Same comment for the catch clause.

Reply:
So what you want is to rename `_triggerUpdateDevicesCheckAgain` to something like `_shouldTriggerUpdateDevices`?